### PR TITLE
feat(bar): bar chart should support a max bar width

### LINKED ISSUE #139: Bar chart - Responsive mode
#### I am submitting a...

- [X] Feature request
- [ ] Design defect
- [ ] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.8.3
### Issue description
**Expected:**  In responsive mode chart weight is expected to adjust to a widget width (expand along with the widget).  Expected to be aligned within the widget.
**Observed:** Component has fixed size and won't adjust as widget expands. Not aligned within widget.

--------------------------------------------------------------------------------------

**2.  Single bar width**
**Expected:** On the Neutrino site there are examples where charts have different bar width (consistent within the same chart).  Below is the example used as a reference for UI design.
**Observed:**  OMSI bars appear thicker than on the mockup. Single bar width changes in responsive mode (becomes even thicker). Our developer team says that bar appearance has to do with the component.

### LINKED ISSUE #140: Bar chart - Single bar width
#### I am submitting a...

- [X] Feature request
- [ ] Design defect
- [ ] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.8.3

### Issue description
**Expected:** On the Neutrino site there are examples where charts have different bar width (consistent within the same chart).  Below is the example used as a reference for UI design.
**Observed:**  OMSI bars appear thicker than on the mockup. Single bar width changes in responsive mode (becomes even thicker). Our developer team says that bar appearance has to do with the component.

### PR DESCRIPTION
**Pull request by cal-smith**
closes #139 
closes #140 

Right now it's just applied to normal bars, but the process should be the same for stacked bars.

Grouped bars will be a bit of a different beast - I'm not sure this entirely applies to them anyway.

### CODE REVIEW

File: packages/core/demo/demo-data/bar.ts [hunk 1]
```diff
@@ -107,7 +107,8 @@ export const simpleBarOptions = {
 		}
 	},
 	legendClickable: true,
-	containerResizable: true
+	containerResizable: true,
+	maxWidth: 50
```
Comment (theiliad) [inline packages/core/demo/demo-data/bar.ts]: Is this for each bar?

This should become:
```js
{
   bars: {
	  maxWidth: 50
   }
}
```

File: packages/core/src/configuration.ts [hunk 2]
```diff
@@ -166,7 +166,8 @@ export const bars = {
 	spacing: {
 		bars: 0.2,
 		datasets: 0.25
-	}
+	},
+	maxWidth: -1
```
Comment (theiliad) [inline packages/core/src/configuration.ts]: Why -1? Why not `null`?

Comment (cal-smith) [reply packages/core/src/configuration.ts]: Just needed an empty value, we can use `null` for sure 👍

File: packages/core/demo/demo-data/bar.ts (hunk 1 above)
Comment (cal-smith) [reply packages/core/demo/demo-data/bar.ts]: sure

**Comment by cal-smith** [comment]
Part of #139 seems to be about some of the extra space reserved for the legend ... It's not immediately obvious to me how the chart is sized, so I'm going to consider that out of scope for this PR ... we should have a separate issue/PR for legend placement and chart sizing 👍

**Pull request merged**